server/src/main/java/com/hedera/block/server/verification/VerificationInjectionModule.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification;
+
+import com.hedera.block.server.metrics.MetricsService;
+import com.hedera.block.server.verification.service.BlockVerificationService;
+import com.hedera.block.server.verification.service.BlockVerificationServiceImpl;
+import com.hedera.block.server.verification.service.NoOpBlockVerificationService;
+import com.hedera.block.server.verification.session.BlockVerificationSessionFactory;
+import com.hedera.block.server.verification.signature.SignatureVerifier;
+import com.hedera.block.server.verification.signature.SignatureVerifierDummy;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import javax.inject.Singleton;
+
+/**
+ * The module used to inject the verification service and signature verifier into the application.
+ */
+@Module
+public interface VerificationInjectionModule {
+
+    /**
+     * Provides the executor service.
+     *
+     * @return the executor service
+     */
+    @Provides
+    static ExecutorService provideExecutorService() {
+        return ForkJoinPool.commonPool();
```

Comment:
We also shouldn't default _all_ executor services to the fork/join common pool; partly because that pool may be single thread, and also because we might want a more "typical" executor service in many cases.

If we need to inject executor services, in general, we need to add a _Qualifier_, such as `@Named`, to differentiate between different types of executor services.

Reply:
@ata-nas @jsync-swirlds I agree that `all` executor services requests would be fulfilled with this one, I guess I took too much inspiration on how it is done on `services` [codebase](https://github.com/hashgraph/hedera-services/blob/main/hedera-node/hedera-app/src/main/java/com/hedera/node/app/platform/PlatformModule.java#L49). 

That said I agree with you both, good idea on how to inject it too, thanks!!

Reply:
✅ Done!